### PR TITLE
show correct duration on graph view for running task (#8311)

### PR DESCRIPTION
**Pull request by fuyi**
This PR is aim to fix issue https://github.com/apache/airflow/issues/8311

#### Idea:

1. Check if a task is still running.
2. If so, disregard the duration field fetched from backend,
3. calculate duration with current timestamp - ti.start_time

#### Tooltip before fix:

Screenshot 2020-05-01 at 22 11 23

#### Tooltip after fix:

screenshot1

The original bug of not showing duration at all on graph view is not the case on master branch, but, the value shows in the tooltip is not correct, since it is read from database and scheduler doesn't update duration frequently to reflect the actual duration till current timestamp for running tasks.

Note: The bug reported in issue 8311 is on v1.10.10 but this PR is based on master branch. Since there are fundamental change between master and v1.10.10 branch, separate PR might be needed to for v1.10.10.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by jeffolsi** [comment]
If I may point Invalid date is also misleading. 
PR

There is nothing invalid here. It just didn't finish so there is no date in Ended.
It should be also modified to "still running" or "in progress" or something else that won't be considered as problem.

**Comment by fuyi** [comment]
Hi @jeffolsi, Thanks for the comment. I agree with you that "invalid date" is not correct either. I have talked with @XD-DENG and leave this out of the scope of this PR.

**Comment by fuyi** [comment]
Hi @ashb , Do you have any comment for this PR?  : )

**Review by ashb — CHANGES_REQUESTED** [review]
Looks good. Could you also fix/remove the `Ended: invalid date` please?

**Comment by fuyi** [comment]
Hi @ashb, `Ended: invalid date` is now fixed 

Screenshot 2020-06-07 at 15 28 01

File: airflow/www/static/js/task-instances.js [hunk 1]
```diff
@@ -20,12 +20,18 @@
...
 function makeDateTimeHTML(start, end) {
+  // check task ended or not
+  if (end && end instanceof moment) {
+    return (
+      `Started: ${start.format(defaultFormat)} <br> Ended: ${end.format(defaultFormat)} <br>`
+    )
+  }
   return (
-    `Started: ${start.format(defaultFormat)} <br> Ended: ${end.format(defaultFormat)} <br>`
+    `Started: ${start.format(defaultFormat)} <br> Ended: Not ended yet <br>`
```
Comment (feluelle) [inline airflow/www/static/js/task-instances.js:34]: Or maybe don't display `Ended:` at all?
```suggestion
    `Started: ${start.format(defaultFormat)} <br>`
```

**Review by feluelle — APPROVED** [review]
LGTM 👍 (only a suggestion)

File: airflow/www/static/js/task-instances.js (hunk 1 above)
Comment (fuyi) [reply airflow/www/static/js/task-instances.js:34]: Thanks for the comment @feluelle. However, I think it is more informative to keep `Ended: Not ended yet`

**Comment by fuyi** [comment]
@ashb Do you have more comments? Can I get approval from you : )

File: airflow/www/static/js/task-instances.js (hunk 1 above)
Comment (feluelle) [reply airflow/www/static/js/task-instances.js:34]: I am totally fine with it. :)

File: airflow/www/static/js/task-instances.js [hunk 2]
```diff
@@ -80,6 +88,17 @@ export default function tiTooltip(ti, {includeTryNumber = false} = {}) {
...
     tt += `Started: ${escapeHtml(ti.start_date)}<br>`;
   }
+  // Calculate duration on the fly if task instance is still running
+  if(ti.state === "running") {
+    let start_date;
+    if (ti.start_date instanceof moment) {
+      start_date = ti.start_date
+    } else {
+      start_date = moment(ti.start_date)
+    }
```
Comment (ashb) [inline airflow/www/static/js/task-instances.js]:
```suggestion
    let start_date = ti.start_date instanceof moment ? ti.start_date : moment(ti.start_date);
```

**Review by ashb — COMMENTED** [review]
Minor comment, and could you please update the screenshot in the PR?

**Comment by fuyi** [comment]
Thanks for comment @ashb , The screenshot is updated now

**Review by ashb — APPROVED** [review]

**Pull request merged**